This patch reverts FFmpeg commit 99042c2bf6cc79006036502a6abbec5e51f73673
(avcodec/h264_slice: don't copy frame data during error concealment) which
broke error concealment with VDPAU.
--- libavcodec/h264_slice.c.orig	2021-09-23 00:08:25.824670926 +0200
+++ libavcodec/h264_slice.c	2021-09-23 02:15:54.725778422 +0200
@@ -1603,15 +1603,14 @@
                 ff_thread_await_progress(&prev->tf, INT_MAX, 0);
                 if (prev->field_picture)
                     ff_thread_await_progress(&prev->tf, INT_MAX, 1);
-                ff_thread_release_buffer(h->avctx, &h->short_ref[0]->tf);
-                h->short_ref[0]->tf.f = h->short_ref[0]->f;
-                ret = ff_thread_ref_frame(&h->short_ref[0]->tf, &prev->tf);
-                if (ret < 0)
-                    return ret;
+                av_image_copy(h->short_ref[0]->f->data,
+                              h->short_ref[0]->f->linesize,
+                              (const uint8_t **)prev->f->data,
+                              prev->f->linesize,
+                              prev->f->format,
+                              prev->f->width,
+                              prev->f->height);
                 h->short_ref[0]->poc = prev->poc + 2U;
-                ff_thread_report_progress(&h->short_ref[0]->tf, INT_MAX, 0);
-                if (h->short_ref[0]->field_picture)
-                    ff_thread_report_progress(&h->short_ref[0]->tf, INT_MAX, 1);
             } else if (!h->frame_recovered && !h->avctx->hwaccel)
                 ff_color_frame(h->short_ref[0]->f, c);
             h->short_ref[0]->frame_num = h->poc.prev_frame_num;
